### Data Preprocessing - Combined Dataset

In [2]:
# Importing necessary libraries
import pandas as pd
import numpy as np

In [13]:
# Reading the dataset
data = pd.read_csv('data/dataset.csv')
data.head()

domain         designation   age     degree  \
0    web development using python django   software engineer  22.0  No Degree   
1  web development using node.js express   software engineer  21.0     Degree   
2    web development using python django    nodejs developer  22.0     Degree   
3                             mern stack  software developer  23.0  No Degree   
4    web development using python django   software engineer  22.0  No Degree   

  cs_bg experience  avg_time holidays  week_back       city  ... linux  sas  \
0    No         No       9.0       No        0.0  bangalore  ...     0    0   
1    No         No      12.0       No        2.0        wfh  ...     0    0   
2   Yes         No       9.0      Yes        4.0  bangalore  ...     0    0   
3    No         No      11.0      Yes        7.0    chennai  ...     0    0   
4    No         No      10.0       No        4.0  bangalore  ...     0    0   

   vercel  redux  api  bootstrap  mongoose  tailwind  docker  salary  
0       0      0    0          0         0         0       0    3.00  
1       0      0    0          0         0         0       0    2.16  
2       0      0    0          0         0         0       0    2.40  
3       0      0    0          0         0         0       0    3.55  
4       0      0    0          0         0         0       0    2.40  

[5 rows x 52 columns]

In [8]:
data.columns

Index(['domain', 'designation', 'age', 'degree', 'cs_bg', 'experience',
       'avg_time', 'holidays', 'week_back', 'city', 'graduated', 'flutter',
       'angular', 'python', 'express', 'firebase', 'aws', 'sql', 'javascript',
       'jwt', 'node', 'html', 'ajax', 'hasura', 'communication', 'git', 'jest',
       'mysql', 'react', 'graphql', 'mongodb', 'cnn', 'postman', 'nginx',
       'dart', 'vim', 'django', 'mui', 'typescript', 'opencv', 'jquery',
       'next', 'linux', 'sas', 'vercel', 'redux', 'api', 'bootstrap',
       'mongoose', 'tailwind', 'docker', 'salary'],
      dtype='object')

In [9]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 669 entries, 0 to 668
Data columns (total 52 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   domain         669 non-null    object 
 1   designation    669 non-null    object 
 2   age            669 non-null    float64
 3   degree         669 non-null    object 
 4   cs_bg          669 non-null    object 
 5   experience     669 non-null    object 
 6   avg_time       669 non-null    float64
 7   holidays       669 non-null    object 
 8   week_back      669 non-null    float64
 9   city           669 non-null    object 
 10  graduated      669 non-null    object 
 11  flutter        669 non-null    int64  
 12  angular        669 non-null    int64  
 13  python         669 non-null    int64  
 14  express        669 non-null    int64  
 15  firebase       669 non-null    int64  
 16  aws            669 non-null    int64  
 17  sql            669 non-null    int64  
 18  javascript

We can drop the column `city` as it has misentries from imputation.

In [14]:
# Dropping the column city
data = data.drop('city', axis=1)
data.head()

domain         designation   age     degree  \
0    web development using python django   software engineer  22.0  No Degree   
1  web development using node.js express   software engineer  21.0     Degree   
2    web development using python django    nodejs developer  22.0     Degree   
3                             mern stack  software developer  23.0  No Degree   
4    web development using python django   software engineer  22.0  No Degree   

  cs_bg experience  avg_time holidays  week_back graduated  ...  linux  sas  \
0    No         No       9.0       No        0.0       Yes  ...      0    0   
1    No         No      12.0       No        2.0       Yes  ...      0    0   
2   Yes         No       9.0      Yes        4.0       Yes  ...      0    0   
3    No         No      11.0      Yes        7.0       Yes  ...      0    0   
4    No         No      10.0       No        4.0       Yes  ...      0    0   

   vercel  redux  api  bootstrap  mongoose  tailwind  docker  salary  
0       0      0    0          0         0         0       0    3.00  
1       0      0    0          0         0         0       0    2.16  
2       0      0    0          0         0         0       0    2.40  
3       0      0    0          0         0         0       0    3.55  
4       0      0    0          0         0         0       0    2.40  

[5 rows x 51 columns]

In [15]:
data.shape

(669, 51)

We have 669 rows and 51 columns

### Encoding Categorical Columns

1. Using **Label encoding** for `degree` column because it has a meaningful ordinal relationship

In [17]:
# Importing label encoder from sklearn
from sklearn.preprocessing import LabelEncoder

# Create an instance of LabelEncoder
label_encoder = LabelEncoder()

In [20]:
pre_data = data.copy()

pre_data['degree_encoded'] = label_encoder.fit_transform(pre_data['degree'])
pre_data.drop('degree', axis=1, inplace=True)
pre_data.head()

domain         designation   age cs_bg  \
0    web development using python django   software engineer  22.0    No   
1  web development using node.js express   software engineer  21.0    No   
2    web development using python django    nodejs developer  22.0   Yes   
3                             mern stack  software developer  23.0    No   
4    web development using python django   software engineer  22.0    No   

  experience  avg_time holidays  week_back graduated  flutter  ...  sas  \
0         No       9.0       No        0.0       Yes        0  ...    0   
1         No      12.0       No        2.0       Yes        0  ...    0   
2         No       9.0      Yes        4.0       Yes        0  ...    0   
3         No      11.0      Yes        7.0       Yes        0  ...    0   
4         No      10.0       No        4.0       Yes        0  ...    0   

   vercel  redux  api  bootstrap  mongoose  tailwind  docker  salary  \
0       0      0    0          0         0         0       0    3.00   
1       0      0    0          0         0         0       0    2.16   
2       0      0    0          0         0         0       0    2.40   
3       0      0    0          0         0         0       0    3.55   
4       0      0    0          0         0         0       0    2.40   

   degree_encoded  
0               2  
1               0  
2               0  
3               2  
4               2  

[5 rows x 51 columns]

In [21]:
pre_data['degree_encoded'].value_counts()

2    338
0    328
1      3
Name: degree_encoded, dtype: int64

##### Legend - degree_encoded
2 - No degree

0 - Degree

1 - Diploma